streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -111,6 +113,7 @@
 
     private final StringSerializer stringSerializer = new StringSerializer();
     private final ByteArraySerializer byteArraySerializer = new ByteArraySerializer();
+    private final UUID processId = UUID.randomUUID();
```

Comment:
Why do we need to introduce this one?

Reply:
StreamsProducer requires the processID to be non-null, but only for eos-v2 for some reason

Reply:
> only for eos-v2 for some reason

That is weird -- if `StreamsProducer` requires the `processID` it should have required it for `_beta` already? Would be good to understand -- maybe we unmasked a bug?